tests/e2e/logging/test_openshift-logging.py

```diff
@@ -0,0 +1,148 @@
+"""
+This file contains the testcases for openshift-logging
+"""
+
+import pytest
+import logging
+
+from tests import helpers
+from ocs_ci.ocs.resources.pod import get_all_pods, get_pod_obj
+from ocs_ci.ocs import constants
+from ocs_ci.utility import deployment_openshift_logging as obj
+from ocs_ci.utility.uninstall_openshift_logging import uninstall_cluster_logging
+from ocs_ci.framework.testlib import E2ETest, tier1
+from ocs_ci.utility.retry import retry
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.fixture()
+def test_fixture(request):
+    """
+    Setup and teardown
+
+    * The setup will deploy openshift-logging in the cluster
+
+    * The teardown will uninstall cluster-logging from the cluster
+
+    """
+
+    def finalizer():
+        teardown(sc_obj, cbp_obj)
+
+    request.addfinalizer(finalizer)
+
+    # Deploys elastic-search operator on the project openshift-operators-redhat
+    obj.create_namespace(yaml_file=constants.EO_NAMESPACE_YAML)
+    assert obj.create_elasticsearch_operator_group(
+        yaml_file=constants.EO_OG_YAML,
+        resource_name='openshift-operators-redhat'
+    )
+    assert obj.set_rbac(
+        yaml_file=constants.EO_RBAC_YAML, resource_name='prometheus-k8s'
+    )
+    assert obj.create_elasticsearch_subscription(constants.EO_SUB_YAML)
+
+    # Deploys cluster-logging operator on the project openshift-logging
+    obj.create_namespace(yaml_file=constants.CL_NAMESPACE_YAML)
+    assert obj.create_clusterlogging_operator_group(
+        yaml_file=constants.CL_OG_YAML
+    )
+    assert obj.create_clusterlogging_subscription(
+        yaml_file=constants.CL_SUB_YAML
+    )
+
+    # Creates instance for cluster-logging
+    cbp_obj = helpers.create_ceph_block_pool()
+    sc_obj = helpers.create_storage_class(
+        interface_type=constants.CEPHBLOCKPOOL,
+        interface_name=cbp_obj.name,
+        secret_name=constants.DEFAULT_SECRET,
+        reclaim_policy="Delete"
+    )
+    assert sc_obj, f"Failed to create storage class"
+    assert obj.create_instance_in_clusterlogging(sc_name=sc_obj.name)
+
+    # Check the health of the cluster-logging
+    assert obj.check_health_of_clusterlogging()
+
+
+def teardown(sc_obj, cbp_obj):
+    """
+    The teardown will uninstall the openshift-logging from the cluster
+    """
+    sc_obj.delete()
+    cbp_obj.delete()
+    uninstall_cluster_logging()
+
+
+@pytest.mark.usefixtures(
+    test_fixture.__name__
+)
+class TestLogging_in_EFK_stack(E2ETest):
+    """
+    The class contains the testcases related to openshift-logging
+    """
+    @pytest.fixture()
+    def create_pvc_and_deploymentconfig_pod(self, request, pvc_factory):
```

Comment:
I suggest setting the pod and pvc as class attributes with self.pod_obj and self.pvc_obj instead of returning them.
Then you can mark this fixture with @pytest.fixture(autouse=True) without passing it as a test function arg

Reply:
@ebenahar this I think is correct use of fixture, we want to return objects from fixture as documented.